analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -125,38 +125,53 @@ public class SymbolReferenceAnalyzerTest
         public void Verify(string fileName, Action<IReadOnlyList<SymbolReferenceInfo.Types.SymbolReference>> verifyReference)
         {
             var testRoot = Root + TestContext.TestName;
-            UtilityAnalyzerBase analyzer = fileName.EndsWith(".cs")
-                ? new TestSymbolReferenceAnalyzer_CS(testRoot)
-                : new TestSymbolReferenceAnalyzer_VB(testRoot);
+            UtilityAnalyzerBase mainAnalyzer = fileName.EndsWith(".cs")
```

Comment:
what's the difference between `mainAnalyzer` and `testAnalyzer`?

Also, if the test scenario is going to fail, all tests are going to fail. This is not a great test setup. A possible refactoring would be to use `DataRow` and merge all `Verify_*_CS` methods into a `Verify_ProductCode_CS` with given parameters.

And then you could have a `Verify_TestCode_CS` with the same parameters, but invoking for test code

Reply:
But feel free to leave the tests as they are if you think it's ok and no refactoring is necessary. I don't have a strong opinion on this one.